mlflow/store/artifact/unity_catalog_oss_models_artifact_repo.py

```diff
@@ -0,0 +1,158 @@
+from mlflow.utils._unity_catalog_utils import (
+    get_artifact_repo_from_storage_info,
+    get_full_name_from_sc,
+)
+from mlflow.utils.oss_utils import get_oss_host_creds
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.rest_utils import (
+    _REST_API_PATH_PREFIX,
+    _UC_OSS_REST_API_PATH_PREFIX,
+    call_endpoint,
+    extract_api_info_for_service,
+)
+from mlflow.utils.uri import (
+    _OSS_UNITY_CATALOG_SCHEME,
+    get_databricks_profile_uri_from_artifact_uri,
+    get_db_info_from_uri,
+    is_oss_unity_catalog_uri,
+)
+
+_METHOD_TO_INFO = extract_api_info_for_service(UcModelRegistryService, _REST_API_PATH_PREFIX)
+_METHOD_TO_INFO_OSS = extract_api_info_for_service(
+    UnityCatalogService, _UC_OSS_REST_API_PATH_PREFIX
+)
+
+
+class UnityCatalogOSSModelsArtifactRepository(ArtifactRepository):
+    """
+    Performs storage operations on artifacts controlled by a Unity Catalog model registry
+
+    Temporary scoped tokens for the appropriate cloud storage locations are fetched from the
+    remote backend and used to download model artifacts.
+
+    The artifact_uri is expected to be of the form `models:/<model_name>/<model_version>`
+
+    Note : This artifact repository is meant is to be instantiated by the ModelsArtifactRepository
+    when the client is pointing to a Unity Catalog model registry.
+    """
+
+    def __init__(self, artifact_uri, registry_uri):
+        if not is_oss_unity_catalog_uri(registry_uri):
+            raise MlflowException(
+                message="Attempted to instantiate an artifact repo to access models in the "
+                f"OSSS Unity Catalog with non-Unity Catalog registry URI '{registry_uri}'. "
+                f"Please specify a Unity Catalog registry URI of the "
+                f"form '{_OSS_UNITY_CATALOG_SCHEME}[://profile]', e.g. by calling "
+                f"mlflow.set_registry_uri('{_OSS_UNITY_CATALOG_SCHEME}') if using the "
+                f"MLflow Python client",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        super().__init__(artifact_uri)
+        from mlflow.tracking.client import MlflowClient
+
+        registry_uri_from_artifact_uri = get_databricks_profile_uri_from_artifact_uri(
+            artifact_uri, result_scheme=_OSS_UNITY_CATALOG_SCHEME
+        )
+        if registry_uri_from_artifact_uri is not None:
+            registry_uri = registry_uri_from_artifact_uri
+        _, key_prefix = get_db_info_from_uri(registry_uri)  # TODO: Ask Kris what to do here
+        if key_prefix is not None:
+            raise MlflowException(
+                "Remote model registry access via model URIs of the form "
+                "'models://<scope>@<prefix>/<model_name>/<version_or_stage>' is unsupported for "
+                "models in the Unity Catalog. We recommend that you access the Unity Catalog "
+                "from the current Databricks workspace instead."
+            )
```

Comment:
let's leave this out for now, once we know more about the format of the URI we can build this validation / parsing into the __init__ function

Reply:
Should we do the URI validation/parsing in this CL itself? I think once we handle the validation/parsing, we won't have to reconfigure all the other methods that currently use the client and registry_uri value. So, taking these lines out and then reconfiguring other functions to skip/not use those values might not make sense. Let me know what you think though. If it will take too long to know about the format of the URI, we can take this out and then recongifure other methods as needed.